pkg/cmd/job/job_kill.go

```diff
@@ -0,0 +1,58 @@
+package job
+
+import (
+	"fmt"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/spf13/cobra"
+)
+
+// newCmdJobKill kills a run of a job.
+func newCmdJobKill(ctx api.Context) *cobra.Command {
+	var all bool
+	cmd := &cobra.Command{
+		Use:   "kill <job-id> [<run-id>]",
+		Short: "kill a job",
+		Args:  cobra.RangeArgs(1, 2),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 2 && all == true {
+				return fmt.Errorf("cannot accept both a run-id and the --all option")
+			}
+			if len(args) == 1 && all == false {
+				return fmt.Errorf("run-id must be specified or --all option must be set")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := metronomeClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			var deadpool []string
+			if len(args) == 1 && all == true {
```

Comment:
Checking all is unnecessary here as there is a pre-check. I'd fine it more clear if there were an early exit path for a specific run-id and a single loop for the `--all` flag, as reading through the shared deadpool slice and the 2 loops is harder to grasp , like this for example : 
``` go
if len(args) == 2 {
	return client.Kill(args[0], args[1])
}

runs, err := client.Runs(args[0])
if err != nil {
	return err
}
for _, run := range runs {
	err = client.Kill(args[0], run.ID)
	if err != nil {
		return err
	}
}
```